exchanges/huobi/huobi_wrapper.go

```diff
@@ -2304,3 +2309,119 @@ func (h *HUOBI) IsPerpetualFutureCurrency(a asset.Item, _ currency.Pair) (bool,
 func (h *HUOBI) UpdateOrderExecutionLimits(_ context.Context, _ asset.Item) error {
 	return common.ErrNotYetImplemented
 }
+
+// GetOpenInterest returns the open interest rate for a given asset pair
+func (h *HUOBI) GetOpenInterest(ctx context.Context, k ...key.PairAsset) ([]futures.OpenInterest, error) {
+	for i := range k {
+		if k[i].Asset != asset.Futures && k[i].Asset != asset.CoinMarginedFutures {
+			// avoid API calls or returning errors after a successful retrieval
+			return nil, fmt.Errorf("%w %v %v", asset.ErrNotSupported, k[i].Asset, k[i].Pair())
+		}
+	}
+	if len(k) == 0 {
```

Comment:
Suggestion: What if we instead checked for `len(k) == 1` and then we do the single call. Anything else `0 or above 1` we can assume a batch call which we can match. This just reduces the multiple rest requests if individuals are defined.

Reply:
I have actioned this where allowable, along with the other comments . May it bring you joy